### PR TITLE
add CFP-28175: BGPv2 API changes

### PR DESCRIPTION
**Pull request by hacktivist123**
This PR stores [CFP28175](https://docs.google.com/document/d/1mqmzlDeVKmNjsnDnnHTeoiW6JcItYMyDyuOM7SYoDi4/edit) in Git for future reference

Authors of the CFP: Harsimran Pabla (@harsimran-pabla), Michael Kashin (@networkop)

### CODE REVIEW

File: cilium/CFP-28175-BGP-v2-API-changes.md [hunk 1]
```diff
@@ -0,0 +1,895 @@
...
+	// +kubebuilder:validation:Optional
+	LocalAddress string `json:"localAddress,omitempty"`
+
+	// LocalPort is port to use for this peer.
+	// +kubebuilder:validation:Optional
+	LocalPort int32 `json:"localPort,omitempty"`
+}
+```
+
+## Backward Compatibility
```
Comment (harsimran-pabla) [inline cilium/CFP-28175-BGP-v2-API-changes.md:774]: This section should be removed, we are reconsidering how migration from v1->v2 will happen. Alternate approach is something like this 
- Provide a CLI which consumes CiliumBGPPeeringPolicy and outputs equivalent BGPv2 CRDs.
- New CRDs can be applied by user
- Enable BGPv2 via feature flag - helm upgrade.

File: cilium/CFP-28175-BGP-v2-API-changes.md [hunk 2]
```diff
@@ -0,0 +1,895 @@
...
+
+**Consumer**
+
+This resource will be consumed by the BGP controller running inside the Cilium agent.
+
+**Status**
+
+No status field planned for this resource.
+
+```go
```
Comment (harsimran-pabla) [inline cilium/CFP-28175-BGP-v2-API-changes.md:237]: There has been update to BGP Advertisement CRD, see [this](https://github.com/cilium/cilium/pull/30902) for more details. I don't know if this CFP need to be upto date with current development as things are always changing slightly.

**Review by harsimran-pabla — COMMENTED** [review]
Thanks a lot @hacktivist123 for translating CFP from doc to this repo. Really appreciate it 🙏 

Currently, we are in development phase of BGPv2. Actual implementation is changing few details ( overall design is still the same ). I have added comments where things have deviated.

**Comment by joestringer** [comment]
Perhaps it would make sense for @harsimran-pabla or other BGP folks to take over this PR? Not sure if that's possible without closing it and having one of you to rebase.

**Comment by hacktivist123** [comment]
@joestringer @harsimran-pabla what do you suggest as the next step for this PR?

What do we need to do to get this merged?

**Comment by harsimran-pabla** [comment]
> What do we need to do to get this merged?

@hacktivist123  The document needs some updates, you can address the comments which I added in this PR. If you'd prefer I can take over this PR as well and push the fixes. I think we can merge it after those changes.

**Comment by hacktivist123** [comment]
Hey @harsimran-pabla you can just take over the PR as it’ll be more efficient since you’re more knowledgeable on this partially CFP. I can close this PR, is that fine?

**Comment by harsimran-pabla** [comment]
> Hey @harsimran-pabla you can just take over the PR as it’ll be more efficient since you’re more knowledgeable on this partially CFP. I can close this PR, is that fine?

Yes, feel free to close it. I will reopen it will latest information.

**Pull request closed without merging**